Ghidra/Processors/ARM/data/languages/ARMneon.sinc

```diff
@@ -1814,6 +1814,86 @@ extImm: "#"^thv_c0811	is TMode=1 & thv_c0811	{ tmp:1 = thv_c0811; export tmp; }
 	Qd = VectorHalvingSubtract(Qn,Qm,esize2021,udt);
 }
 
+#######
+# VFMA VFMS VFNMA and VFNMS
+#
+
+@if defined(VFPv4)
+
+:vfma^COND^".f16" Sd,Sn,Sm	is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=1 & c0606=0 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=1 & thv_c0606=0 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = zext(Sd:2 f+ (Sn:2 f* Sm:2));
+}
+
+:vfma^COND^".f32" Sd,Sn,Sm	is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=2 & c0606=0 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=2 & thv_c0606=0 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = Sd f+ (Sn f* Sm);
+}
+
+:vfma^COND^".f64" Dd,Dn,Dm	is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=3 & c0606=0 & c0404=0) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=3 & thv_c0606=0 & thv_c0404=0)) & Dm & Dn & Dd
+{
+	Dd = Dd f+ (Dn f* Dm);
+}
+
+:vfms^COND^".f16" Sd,Sn,Sm	is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=1 & c0606=1 & c0404=0) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=1 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = zext(Sd:2 f- (Sn:2 f* Sm:2));
+}
+
+:vfms^COND^".f32" Sd,Sn,Sm	is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=2 & c0606=1 & c0404=0) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=2 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = Sd f- (Sn f* Sm);
+}
+
+:vfms^COND^".f64" Dd,Dn,Dm  is ( ( $(AMODE) & COND & c2327=0x1d & c2021=2 & c1011=2 & c0809=3 & c0606=1 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=2 & thv_c1011=2 & thv_c0809=3 & thv_c0606=1 & thv_c0404=0)) & Dm & Dn & Dd
+{
+	Dd = Dd f- (Dn f* Dm);
+}
+
+:vfnma^COND^".f16" Sd,Sn,Sm is ( ( $(AMODE) & COND & c2327=0x1d & c2021=1 & c1011=2 & c0809=1 & c0606=1 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=1 & thv_c1011=2 & thv_c0809=1 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = zext(Sd:2 f+ ((0 f- Sn:2) f* Sm:2));
+}
+
+:vfnma^COND^".f32" Sd,Sn,Sm is ( ( $(AMODE) & COND & c2327=0x1d & c2021=1 & c1011=2 & c0809=2 & c0606=1 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=1 & thv_c1011=2 & thv_c0809=2 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = Sd f+ ((0 f- Sn) f* Sm);
```

Comment:
This calculation isn't right. Basedo n the manual, it should be:
`Sd = (f- Sd) f+ ((f- Sn) f* Sm);`
Similarly for the other vfnma instructions.

Reply:
fixed in 808143cd606e1109bb354ee8873bd9734f5ff4a9